### PR TITLE
*: upgrade etcd-operator to 0.4.0

### PR DESCRIPTION
**Pull request by hongchaodeng**

### CODE REVIEW

**Comment by hongchaodeng** [comment]
coreosbot test e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
Manually Verified both bootstrap and recovery processes.

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
Previously was having some issues and tested wrong branch.
Now fixed.

**Comment by hongchaodeng** [comment]
coreosbot run e2e

File: pkg/asset/internal/templates.go [hunk 1]
```diff
@@ -1040,7 +1040,7 @@ var EtcdTPRTemplate = []byte(`{
       "static": {
         "member": {
           "peerSecret": "etcd-member-peer-tls",
-          "clientSecret": "etcd-member-client-tls"
+          "serverSecret": "etcd-member-client-tls"
```
Comment (aaronlevy) [inline pkg/asset/internal/templates.go]: nit: in the etcd-operator docs the secret name is slightly different (don't have "member"). https://github.com/coreos/etcd-operator/blob/master/doc/user/cluster_tls.md#static-cluster-tls-policy

but that can be a follow up too

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
coreosbot run e2e

**Comment by hongchaodeng** [comment]
All passed :)

**Pull request merged**